src/main/java/de/tum/in/test/api/structural/testutils/ClassNameScanner.java

```diff
@@ -297,17 +297,22 @@ private void walkProjectFileStructure(String assignmentFolderName, File node,
 
 			var className = fileNameComponents[fileNameComponents.length - 2];
 			var packageName = node.getPath().substring(0, node.getPath().indexOf("." + fileExtension));
-			packageName = packageName.substring(
-					packageName.indexOf(assignmentFolderName) + assignmentFolderName.length() + 1,
-					packageName.lastIndexOf(File.separator + className));
-			packageName = packageName.replace(File.separatorChar, '.');
+			int packageStartIndex = packageName.indexOf(assignmentFolderName) + assignmentFolderName.length() + 1;
+			int packageEndIndex = packageName.lastIndexOf(File.separator + className);
+
+			if (packageStartIndex + 1 < packageEndIndex) {
+				packageName = packageName.substring(packageStartIndex, packageEndIndex).replace(File.separatorChar,
+						'.');
+				if (packageName.charAt(0) == '.')
+					packageName = packageName.substring(1);
+			} else {
+				packageName = ""; // Default package
+			}
```

Comment:
It might be worth replacing this index-substring construct by `Path`. Something like
```java
Path packagePath = Path.of(assignmentFolderName).relativize(Path.of(packageName))
// and maybe
String qualifiedClassName = StreamSupport.stream(packagePath.spliterator(), false).map(Object::toString).collect(Collectors.joining("."));
```
This yields `[Pinguin, de.tum.eist.Pattern]` for `[assignment\src\Pinguin, assignment\src\de\tum\eist\Pattern]`

Reply:
Thanks for the suggestion. I tested your code and it still works and looks a lot nicer